sway-core/src/parse_tree/declaration/function.rs

```diff
@@ -50,7 +50,7 @@ impl FunctionDeclaration {
             .unwrap_or(false)
         {
             let _ = signature.next();
-            Purity::Impure
+            Purity::Reads // TODO: add attributes to parser instead of `impure` keywords.  Or not.
```

Comment:
Wait why is this TODO here?

Reply:
This is about the old Pest parser, which Alex already has a branch started for removing.  So this is an almost definitely WONTDO.

Reply:
Ahhh got it